src/mod_muc_db_rdbms.erl

```diff
@@ -0,0 +1,255 @@
+-module(mod_muc_db_rdbms).
+-include("mod_muc.hrl").
+-include("mongoose_logger.hrl").
+
+-export([init/2,
+         store_room/4,
+         restore_room/3,
+         forget_room/3,
+         get_rooms/2,
+         can_use_nick/4,
+         get_nick/3,
+         set_nick/4,
+         unset_nick/3
+        ]).
+
+-define(ESC(T), mongoose_rdbms:use_escaped_string(mongoose_rdbms:escape_string(T))).
+
+%% Defines which RDBMS pool to use
+%% Parent host of the MUC service
+-type server_host() :: ejabberd:server().
+
+%% Host of MUC service
+-type muc_host() :: ejabberd:server().
+
+%% User's JID. Can be on another domain accessable over FED.
+%% Only bare part (user@host) is important.
+-type client_jid() :: ejabberd:jid().
+
+-type room_opts() :: [{OptionName :: atom(), OptionValue :: term()}].
+
+-type aff() :: atom().
+
+
+-spec init(server_host(), ModuleOpts :: list()) -> ok.
+init(ServerHost, _Opts) ->
+    rdbms_queries:prepare_upsert(ServerHost, muc_nick_upsert, muc_registered,
+                                 [<<"muc_host">>, <<"luser">>, <<"lserver">>, <<"nick">>],
+                                 [<<"nick">>],
+                                 [<<"muc_host">>, <<"luser">>, <<"lserver">>]),
+    ok.
+
+-spec store_room(server_host(), muc_host(), mod_muc:room(), room_opts()) ->
+    ok | {error, term()}.
+store_room(ServerHost, MucHost, RoomName, Opts) ->
+    forget_room(ServerHost, MucHost, RoomName),
```

Comment:
I think it'd be safer to put the`forget_room` inside the `store_room_transaction`. Thanks to that we'll be sure that we have consistent state. Currently it may happen that someone else queries for the room which was removed only to insert it again.